File: exercism/go/anagram/anagram.go
package anagram

import (
	"sort"
	"strings"
)

func Sorted(word string) string {
	s := []rune(word)
	sort.Slice(s, func(i int, j int) bool { return s[i] < s[j] })

	return string(s)
}

func Detect(subject string, candidates []string) []string {

	var values []string

	subject = strings.ToLower(subject)
	test_subject := Sorted(strings.ToLower(subject))

	for _, candidate := range candidates {
		test_candidate := strings.ToLower(candidate)

		if test_subject == Sorted(test_candidate) && subject != test_candidate {
			values = append(values, candidate)
		}
	}

	return values

}

File: exercism/python/anagram/anagram.py
"""
anagram.py
"""


def find_anagrams(word, candidates):
    """
    Given a target word and a set of candidate words, this exercise requests
    the anagram set: the subset of the candidates that are anagrams of the
    target.

        arg word: (int) word to check
        arg candidates: (set) list of candidates
        return: (set) of anagrams
    """

    values = set()
    word = word.casefold()
    word_test = sorted(word.casefold())

    for candidate in candidates:
        test_candidate = candidate.casefold()
        if word_test == sorted(test_candidate) and word != test_candidate:
            values.add(candidate)

    return values
